pkg/kongstate/consumer.go

```diff
@@ -3,10 +3,15 @@ package kongstate
 import (
 	"fmt"
 
+	"github.com/blang/semver"
 	"github.com/kong/go-kong/kong"
 	configurationv1 "github.com/kong/kubernetes-ingress-controller/pkg/apis/configuration/v1"
 )
 
+var (
+	minMTLSCredentialVersion, _ = semver.Make("2.3.2")
```

Comment:
Newer versions of semver have a `MustParse` function which I believe would be preferable here:

```
// MustParse is like Parse but panics if the version cannot be parsed.
func MustParse(s string) Version {
	v, err := Parse(s)
	if err != nil {
		panic(`semver: Parse(` + s + `): ` + err.Error())
	}
	return v
}
```

https://github.com/blang/semver/blob/master/v4/semver.go#L360

```suggestion
	minMTLSCredentialVersion = semver.MustParse("2.3.2")
```

Technically in the `err` case of `Make(v)` if an error occurs, you still receive a `semver.Version` default of:

```
"0.0.0"
```

This is a bit of a nitpick because I think we all expect this will never result in an error anyway, but if we expect that I would submit we might as well choose the _defined_ behavior for the condition we think will never happen, rather than the _undefined_ (or at least _less clear_) one.